### DISCUSSION TITLE
iOS shortcut that syncs your latest workout, for docker with API/webUI setup

### DISCUSSION THREAD

**Discussion by qpop01** [philosowaffle/peloton-to-garmin#455] [Show and tell]
Hi all.

This is my preferred workflow, with manual sync the moment I get off the bike (I like my data in Garmin ASAP for health tracking).

This shortcut will ask on import what URL/IP your p2g is hosted at (i.e. http://rpi.local) - you need to include the leading http://

Once input, it makes an API GET request for the latest workout ID, and then an API post request to sync just that workout.

If you like it @philosowaffle please feel free to add it to the docs. It's a big jump in usability for me over having to go to the webui!

You can download the shortcut from icloud, or import the file manually (attached as a .zip because github doesn't like .shortcut files), or if you'd rather build it yourself, screenshots are attached.

(The standard JSON formatting in POST requests in shortcuts doesn't format the request correctly, so the text file in the shortcut is a workaround).

https://www.icloud.com/shortcuts/58f0c2c099c34682aa2eddbcf5b05ccc

[Peloton To Garmin.shortcut.zip](https://github.com/philosowaffle/peloton-to-garmin/files/10973711/Peloton.To.Garmin.shortcut.zip)

![20230314_210907649_iOS](https://user-images.githubusercontent.com/37111291/225137270-3e84b534-15c9-4dd4-8355-4ad71b86a397.png)
![20230314_210907603_iOS](https://user-images.githubusercontent.com/37111291/225137262-cbf8b40e-e8d7-4dd8-b5e5-6b82afe31a7c.png)

**Comment by klspindler** [comment]
I am new to Github and Docker, and only stubbled upon P2G today after Garmin forced me to use 2FA.  Would you mind sharing how to determine where P2G is hosted?  The WebUI is working on my Mac, but I don't seem to be able to connect from my iPhone.  Thank you!

**Comment by yetdog** [comment]
This is incredibly useful; thank you!
